scripts/tile.py

```diff
@@ -84,22 +78,18 @@ def tiler(stack, date, mgrs):
                 y_end = y_start + TILE_SIZE
 
                 # Select the subset of data for the current tile
-                tile = stack.sel(
-                    x=slice(
-                        stack.x.values[x_start],
-                        stack.x.values[x_end]
-                        + np.sign(stack.rio.transform()[4]) * EPSILON,
-                    ),
-                    y=slice(
-                        stack.y.values[y_start],
-                        stack.y.values[y_end]
-                        + np.sign(stack.rio.transform()[0]) * EPSILON,
-                    ),
-                )
+                parts = [part[:, y_start:y_end, x_start:x_end] for part in stack]
+
+                # Only concat here to save memory, it converts S2 data to float
+                tile = xr.concat(parts, dim="band").rename("tile")
```

Comment:
@yellowcap or @lillythomas, can you double check the Sentinel-1 pixel values? I've been getting NaN loss values, and tracked it down to the Sentinel-1 VV and VH bands being all NaNs. Sentinel-2 and DEM bands are ok. Sample code:

```python
import os
import matplotlib.pyplot as plt
import rioxarray
import xarray as xr

os.environ["AWS_PROFILE"] = "xxx"

da = rioxarray.open_rasterio(
    filename="s3://clay-tiles-01/01/56HKH/2022-06-18/claytile-56HKH-2022-06-18-01-1.tif"
)
da = da.compute()

# %%
# Plot different bands
da.sel(band=1).plot.imshow()  # Blue
plt.show()
da.sel(band=10).plot.imshow()  # SWIR2
plt.show()
da.sel(band=11).plot.imshow()  # VH
plt.show()
da.sel(band=12).plot.imshow()  # VV
plt.show()
da.sel(band=13).plot.imshow()  # DEM
plt.show()
```

![Blue band](https://github.com/Clay-foundation/model/assets/23487320/4d767e8f-98fe-4924-93b1-a225191323fd)
![SWIR2 band](https://github.com/Clay-foundation/model/assets/23487320/8567c9ef-411e-4206-b6e2-8b4aa42e219a)
![VH band](https://github.com/Clay-foundation/model/assets/23487320/955475fd-9c9b-483c-8180-86bef58f4e59)
![VV band](https://github.com/Clay-foundation/model/assets/23487320/7ff10505-1b44-48ce-8bb2-bb0b0a3a5784)
![DEM band](https://github.com/Clay-foundation/model/assets/23487320/3db2b6b0-bf4c-4adc-859e-3ad632953ce6)

Reply:
Tried a few other tiles in other locations like MGRS 29SMD over Lisbon, and it seems like the DEM can come up with NaNs as well (in places over the ocean?). So maybe check through all those bands.

Reply:
Good catch! Looks like the [nodata filter](https://github.com/Clay-foundation/model/blob/f48c01520099d7ef7ea916008573be3959c176a9/scripts/tile.py#L27) is missing checks on S1 and DEM. That should be easy to correct.

Reply:
Ok, look like Lilly will be fixing this in #60.

Reply:
converted to issue for tracking https://github.com/Clay-foundation/model/issues/68